builder/dockerfile/builder.go

```diff
@@ -196,6 +199,11 @@ func (b *Builder) dispatchDockerfileWithCancellation(dockerfile *parser.Result)
 		// TODO: get this from dispatch
 		imageID = b.image
 
+		if b.Aux != nil {
+			if err := b.Aux.Emit(types.BuildResult{ID: imageID}); err != nil {
+				return "", err
+			}
+		}
```

Comment:
Oh, that's possible. Lets see what @tonistiigi says.

(if the answer is the ID for each stage then I'd appreciate a pointer to where those Id are made available or how I can distinguish the end of a stage from a regular layer)